### DISCUSSION TITLE
Moes 2 gang zigbee 3 dimmer switch MS-105BZ not being fully recognised by Tasmota

### DISCUSSION THREAD

**Discussion by jpadie** [arendst/Tasmota#21265] [Support request]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

Pairing between tasmota zigbee bridge pro on a sonoff zigbee bridge pro and the MS105BZ works ok.  pairing is effected by 10 slow presses on a rocker switch connected to the switch terminals of the modules. 

however tasmota seems to see only one gang and control of either is not assured (not had much luck yet).

discussion on Discord moved to github at Stephan's request.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff Zigbee Bridge Pro_____
- [x] Tasmota binary firmware version number used: Tasmota 13.4.0(TasmoCompiler-esp32generic)  
  - [ ] Pre-compiled
  - [x ] Self-compiled
- [x] Flashing tools used: web installer for vanilla pre-compiled binary followed by web firmware upload for binary with matter compiled in. 
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
11:05:39.452 CMD: Backlog Template; Module; GPIO 255
11:05:39.455 SRC: WebConsole from 10.2.10.30
11:05:39.457 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 26, Pld -99, Data 'Template; Module; GPIO 255'
11:05:39.488 SRC: Backlog
11:05:39.491 CMD: Grp 0, Cmd 'TEMPLATE', Idx 1, Len 0, Pld -99, Data ''
11:05:39.505 MQT: oc/zbpro/stat/RESULT = {"NAME":"Sonoff Zigbee Pro","GPIO":[0,0,576,0,480,0,0,0,0,1,1,5792,0,0,0,3552,0,320,5793,3584,0,640,608,32,0,0,0,0,0,1,0,0,0,0,0,0],"FLAG":0,"BASE":1}
11:05:39.718 SRC: Backlog
11:05:39.720 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
11:05:39.730 MQT: oc/zbpro/stat/RESULT = {"Module":{"0":"Sonoff Zigbee Pro"}}
11:05:39.968 SRC: Backlog
11:05:39.969 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 3, Pld 255, Data '255'
11:05:39.988 MQT: oc/zbpro/stat/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"576":"LedLink_i"},"GPIO3":{"0":"None"},"GPIO4":{"480":"Buzzer"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"5792":"Zigbee Rst1"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"3552":"Zigbee Tx"},"GPIO20":{"0":"None"},"GPIO21":{"320":"Led_i1"},"GPIO22":{"5793":"Zigbee Rst2"},"GPIO23":{"3584":"Zigbee Rx"},"GPIO24":{"0":"None"},"GPIO25":{"640":"I2C SDA1"},"GPIO26":{"608":"I2C SCL1"},"GPIO27":{"32":"Button1"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}

```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
no rules in use

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
11:07:12.519 MQT: oc/zbpro/stat/STATUS = {"Status":{"Module":0,"DeviceName":"ZigbeePro","FriendlyName":["ZigbeePro"],"Topic":"zbpro","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
11:07:12.526 MQT: oc/zbpro/stat/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/tasmota32-zbbrdgpro.bin","RestartReason":"Software reset CPU","Uptime":"0T15:10:51","StartupUTC":"2024-04-23T18:56:21","Sleep":50,"CfgHolder":4617,"BootCount":12,"BCResetTime":"2024-04-23T10:22:48","SaveCount":34}}
11:07:12.534 MQT: oc/zbpro/stat/STATUS2 = {"StatusFWR":{"Version":"13.4.0(TasmoCompiler-esp32generic)","BuildDateTime":"2024-04-23T10:57:29","Core":"2_0_14","SDK":"4.4.6.240105","CpuFrequency":160,"Hardware":"ESP32-D0WD-V3 v3.1","CR":"412/699"}}
11:07:12.541 MQT: oc/zbpro/stat/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":3,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Orchard Cottage",""],"TelePeriod":300,"Resolution":"55C180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
11:07:12.556 MQT: oc/zbpro/stat/STATUS4 = {"StatusMEM":{"ProgramSize":1781,"Free":74,"Heap":141,"StackLowMark":3,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"1640C8","FlashFrequency":40,"FlashMode":"DIO","Features":["0809","078007DE","005C0001","00000003","00000000","C0000881","40020080","00200000","54000808","80000080","00000004"],"Drivers":"1,2,!4,!8,9,10,12,20,!21,23,24,29,!35,38,!41,50,52,!59,62,!68,82,!121","Sensors":"1,2,127","I2CDriver":""}}
11:07:12.567 MQT: oc/zbpro/stat/STATUS5 = {"StatusNET":{"Hostname":"zbpro-0004","IPAddress":"10.2.10.40","Gateway":"10.2.10.1","Subnetmask":"255.255.255.0","DNSServer1":"****","DNSServer2":"*****","Mac":"E4:65:B8:54:60:04","IP6Global":"*******","IP6Local":"fe80::e665:b8ff:fe54:6004%st1","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"****","DNSServer2":"****","Mac":"00:00:00:00:00:00","IP6Global":"","IP6Local":""},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":16.0}}
11:07:12.574 MQT: oc/zbpro/stat/STATUS6 = {"StatusMQT":{"MqttHost":"********","MqttPort":8883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_546004","MqttUser":"rab2","MqttCount":6,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
11:07:12.583 MQT: oc/zbpro/stat/STATUS7 = {"StatusTIM":{"UTC":"2024-04-24T10:07:12Z","Local":"2024-04-24T11:07:12","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"05:41","Sunset":"19:55"}}
11:07:12.595 MQT: oc/zbpro/stat/STATUS10 = {"StatusSNS":{"Time":"2024-04-24T11:07:12"}}
11:07:12.604 MQT: oc/zbpro/stat/STATUS11 = {"StatusSTS":{"Time":"2024-04-24T11:07:12","Uptime":"0T15:10:51","UptimeSec":54651,"Heap":139,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":6,"Berry":{"HeapUsed":4,"Objects":46},"Wifi":{"AP":1,"SSId":"Orchard Cottage","BSSId":"64:DA:ED:8D:AD:05","Channel":11,"Mode":"11n","RSSI":54,"Signal":-73,"LinkCount":1,"Downtime":"0T00:00:05"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:

console output from pairing and button press attached as text files.

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
pair module MS-105BZ to tasmota. 

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

Expectation is that on/off and dim commands will be available for each gang of the module separately. 

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

discord input from Stephan follows

At first glance, it is unfortunately a Tuya zigbee device. Tuya invented their own protocol over Zigbee and does not follow the Zigbee standards. Let me check
Based on Zigbee2MQTT your device signature is MS-105B not MS-105Z

With a zigbee plugin you should be able to see the first gang. But for the second, I'm afraid it will need to Berry code for this. It's actually a good use case for this. Can you please open a Github discussion instead of Discord? It's easier to follow-up

JPADIE response:  looking at the z2m site, the version that I have is physically similar to both the 105B and 105Z.  the case marking is 105BZ and the power rating is 2x100w.  clicking on the 105B image on the z2m site takes the viewer to a completely different module (https://www.zigbee2mqtt.io/devices/TS0601_dimmer_2.html).  perhaps similar underneath though (given the ts0601 identifier).   the pairing instructions given for each of the 105B and 105Z are different and those for the 105Z work on the module I have.  this page (https://zigbee.blakadder.com/Moes_MS-105B.html) suggests the module is confirmed to work with tasmota, which suggests that it is not the same as the MS-105B despite the identical Manufacturer designation. 

[activity logs.txt](https://github.com/arendst/Tasmota/files/15092808/activity.logs.txt)
[pairing logs.txt](https://github.com/arendst/Tasmota/files/15092809/pairing.logs.txt)

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by jpadie** [comment]
@s-hadinger posted as requested.

**Reply by s-hadinger** [reply]
Thanks, will follow-up here.

**Reply by jpadie** [reply]
i paired the devices to Z2M and it indeed recognises them as a TS0601_dimmer_2 which Z2M believes is a white label variant of the MS105B.  the page is here: https://www.zigbee2mqtt.io/devices/TS0601_dimmer_2.html#tuya-ts0601_dimmer_2.  

however my device is a module, not a 2 gang switch.  obv same stuff inside.  

functionality is as it should be in Z2M.  hope that helps.

**Comment by jpadie** [comment]
@s-hadinger I should have some free desk time this week and wondered if you could share your approach to adding support for tuya type modules in tasmota's ZigBee implementation so that I could start having a go?
